test/framework/src/main/java/org/elasticsearch/test/disruption/NetworkDisruptionTests.java

```diff
@@ -56,6 +56,21 @@ private void checkTwoPartitions(TwoPartitions topology, Set<String> partition1,
         assertTrue(topology.getMajoritySide().size() >= topology.getMinoritySide().size());
     }
 
+    public void testIsolateAll() {
+        Set<String> nodes = generateRandomStringSet(1, 10);
+        NetworkDisruption.DisruptedLinks topology = new NetworkDisruption.IsolateAllNodes(nodes);
+        for (int i = 0; i < 10; i++) {
```

Comment:
yeah , balancing act between speed and chances the test fails if you get something wrong. I also just hate the resulting double loop for a "check all combinations"